### PR TITLE
alarm/uboot-rock64: update u-boot and rkbin versions

### PR DESCRIPTION
**Pull request by yogo1212**
Hello :-)

First submission here, trying to follow the guidelines...
During the attempts to get the display working on my Rock64, I created an updated PKGBUILD for the u-boot package that
- uses newer versions of u-boot and rkbin
- drops the rk3328trust.ini in favour of a corresponding file that is present in the upstream repo anyway
- reads the binary file names containing independent version numbers directly from another INI file that is also in the upstream repository

The package builds and produces a working u-boot.
My display is now working and the changes looked useful to me, so I'm sharing them here.

A build for 2019.07 is pending.
If it works, I'll update the version in this PR.

regards

### CODE REVIEW

**Comment by kmihelich** [comment]
Read them again:
https://github.com/archlinuxarm/PKGBUILDs/blob/master/CONTRIBUTING.md

File: alarm/uboot-rock64/PKGBUILD [hunk 1]
```diff
@@ -13,20 +13,18 @@ license=('GPL')
 backup=('boot/boot.txt' 'boot/boot.scr')
 makedepends=('bc' 'git' 'rockchip-tools')
 install=${pkgname}.install
-_commit_rkbin=f64ded672836b47db3f6a0d04d8ccb4180fb3d55
-source=("ftp://ftp.denx.de/pub/u-boot/u-boot-${pkgver/rc/-rc}.tar.bz2"
+_commit_rkbin=0b8a4e66958b5d3d87fc61fc4bd8e069c79e931a
+source=("git+https://gitlab.denx.de/u-boot/u-boot.git#commit=v${pkgver}"
```
Comment (kmihelich) [inline alarm/uboot-rock64/PKGBUILD]: Leave this as the tarball source.

Comment (yogo1212) [reply alarm/uboot-rock64/PKGBUILD]: yes, sir :stuck_out_tongue: 

squashing this changeset:
```
diff --git a/alarm/uboot-rock64/PKGBUILD b/alarm/uboot-rock64/PKGBUILD
index 46ad52dae..39d0d4cbe 100644
--- a/alarm/uboot-rock64/PKGBUILD
+++ b/alarm/uboot-rock64/PKGBUILD
@@ -4,7 +4,7 @@
 buildarch=8
 
 pkgname=uboot-rock64
-pkgver=2019.04
+pkgver=2019.07
 pkgrel=1
 pkgdesc="U-Boot for Rock64"
 arch=('aarch64')
@@ -14,7 +14,7 @@ backup=('boot/boot.txt' 'boot/boot.scr')
 makedepends=('bc' 'git' 'rockchip-tools')
 install=${pkgname}.install
 _commit_rkbin=0b8a4e66958b5d3d87fc61fc4bd8e069c79e931a
-source=("git+https://gitlab.denx.de/u-boot/u-boot.git#commit=v${pkgver}"
+source=("ftp://ftp.denx.de/pub/u-boot/u-boot-${pkgver/rc/-rc}.tar.bz2"
         "git+https://github.com/rockchip-linux/rkbin.git#commit=$_commit_rkbin"
         'boot.txt'
         'mkscr')
@@ -24,7 +24,7 @@ md5sums=('SKIP'
          '021623a04afd29ac3f368977140cfbfd')
 
 build() {
-  cd u-boot
+  cd u-boot-${pkgver/rc/-rc}
 
   unset CLFAGS CXXFLAGS CPPFLAGS LDFLAGS
 
@@ -34,7 +34,7 @@ build() {
 }
 
 package() {
-  cd u-boot
+  cd u-boot-${pkgver/rc/-rc}
 
   mkdir -p "${pkgdir}/boot"
 

```

Comment (yogo1212) [reply alarm/uboot-rock64/PKGBUILD]: .. and the checksum

Comment (yogo1212) [reply alarm/uboot-rock64/PKGBUILD]: @kmihelich built and tested

**Comment by yogo1212** [comment]
dude, what do you mean by `Read them again`?
would the switch to git have required a separate commit?
git commit message? (that isn't part of the guidelines)
more explanation on the version increment?
or is your issue solved through the review?
i'm having trouble identifying what i should've done better.

and you come across as being a bit condescending.
if you feel i'm wasting your time, first of all
i'm sorry - really not my intention;
then maybe be more explicit about what bothers you or improve the guidelines.

**Comment by kmihelich** [comment]
What part of **squash your commits** is unclear?

Whatever condescension you perceive is well outweighed by the disrespect you are showing for not following the published **rules** for contribution to this repository, and furthered by attacking the people pointing out your failure to comply with them.

**Comment by yogo1212** [comment]
Thank you for telling me now.
I'm unhappy about my last comment.
The guidelines are unclear about when the squashing takes place.
I'm used to the squashing taking place right before the merge or at the reviewers request to make work with intermediate changesets easier.

The first thing I said was that I striving towards understanding the guidelines, which I assume exist to make your life easier.
"Read them again: [resource i know you already have]" is not really different from a "You failed; I'm not going to be of any help.".
It wouldn't have been any more difficult to just say I should squash the commits.

If you're unhappy about helping people creating contributions you can work with, why are you reviewing them?

Anyways - I'm squashing the commits despite the other changes not related to the new u-boot version.
I don't suppose we meet again,
fare well and take care.

**Pull request closed without merging**